File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        mxDigit = ' '
        for i in range(len(num)-2):
            if num[i] == num[i+1] == num[i+2]:
                mxDigit = max(mxDigit, num[i])
        return '' if mxDigit==' ' else mxDigit*3

File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.java
class Solution {
    public String largestGoodInteger(String num) {
        char mxDigit = ' ';
        for(int i=0; i<=num.length()-3; i++){
            if(num.charAt(i)==num.charAt(i+1) && num.charAt(i)==num.charAt(i+2)){
                mxDigit = (char) Math.max(mxDigit, num.charAt(i));
            }
        }
        return mxDigit==' ' ? "" : new String(new char[]{mxDigit, mxDigit, mxDigit});
    }
}